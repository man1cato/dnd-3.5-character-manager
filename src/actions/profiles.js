import database from '../firebase/firebase';


//SET PROFILE IDS IN STORE
export const setProfiles = (profiles) => ({
    type: 'SET_PROFILES',
    profiles
})

//READ PROFILES FROM FIREBASE 
export const startGetProfiles = (uid) => {
    return (dispatch) => {
        return database.ref(`users/${uid}/profiles`).once('value').then((snapshot) => {
            let profiles = snapshot.val()
            if (profiles) {
                const profileIds = Object.keys(profiles)
                profiles = profileIds.map((id) => ({
                    id,
                    name: profiles[id].name,
                    race: profiles[id].race,
                    gender: profiles[id].gender,
                    jobClass: profiles[id].jobClass,
                    alignment: profiles[id].alignment,
                    level: profiles[id].level,
                    iconUrl: profiles[id].iconUrl
                }))
                dispatch(setProfiles(profiles))
            }
        })
    }
}

//REMOVE PROFILES FROM STORE
export const removeProfiles = () => ({
    type: 'REMOVE_PROFILES'
})