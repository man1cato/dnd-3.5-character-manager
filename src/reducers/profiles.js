export default (state = [], action) => {
    switch (action.type) {
        case 'SET_PROFILES':
            return [
                ...action.profiles
            ]
        case 'DELETE_PROFILES':
            return []
        default:
            return state
    }
}
