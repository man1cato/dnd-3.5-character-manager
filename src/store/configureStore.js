import {createStore, combineReducers, applyMiddleware, compose} from 'redux';
import thunk from 'redux-thunk';
import authReducer from '../reducers/auth';
import profileReducer from '../reducers/profile';
import spellsReducer from '../reducers/spells';
import skillsReducer from '../reducers/skills';
import featsReducer from '../reducers/feats';
import specialAbilitiesReducer from '../reducers/specialAbilities';
import itemsReducer from '../reducers/items';
import racesReducer from '../reducers/races';
import jobClassesReducer from '../reducers/jobClasses';


const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default () => {
	const store = createStore(
		combineReducers({
			auth: authReducer,
			profile: profileReducer,
			spells: spellsReducer,
			skills: skillsReducer,
			feats: featsReducer,
			specialAbilities: specialAbilitiesReducer,
			items: itemsReducer,
			races: racesReducer,
			jobClasses: jobClassesReducer
		}),
		composeEnhancers(applyMiddleware(thunk))
	);

	return store;
};
