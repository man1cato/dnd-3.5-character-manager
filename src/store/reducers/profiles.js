export default (state = [], action) => {
    switch (action.type) {
        case 'SET_PROFILES':
            return action.profiles
        case 'REMOVE_PROFILES':
            return []
        default:
            return state
    }
}
