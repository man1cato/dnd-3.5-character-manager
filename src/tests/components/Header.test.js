import React from 'react'
import { Provider } from 'react-redux'
import { StaticRouter, NavLink } from 'react-router-dom'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'

import profiles from '../fixtures/profiles'
import ConnectedHeader, { Header } from '../../components/Header/Header'

const mockStore = configureStore()
const createConnectedWrapper = (state) => mount(
    <Provider store={mockStore(state)}>
        <StaticRouter>
            <ConnectedHeader /> 
        </StaticRouter>
	</Provider>
)

let wrapper

test('should render Header correctly', () => {
    wrapper = shallow(<Header startLogout={() => {}}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should call startLogout on button click', () => {
    const startLogout = jest.fn()
    wrapper = shallow(<Header startLogout={startLogout}/>)
    wrapper.find('button').simulate('click')
    expect(startLogout).toHaveBeenCalled()
})

test('should show select character icon when profiles in store', () => {
    wrapper = createConnectedWrapper({ profiles })
    expect(wrapper.find(NavLink).at(1).props().hidden).toEqual(false)
})