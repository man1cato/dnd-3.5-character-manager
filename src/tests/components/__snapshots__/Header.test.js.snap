// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="container container--header header__content"
  >
    <h2
      className="header__title"
    />
    <button
      className="button button--link"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
